### PR TITLE
Improvements for Payload inspector

### PR DESCRIPTION
**Pull request by ggovi**
In this PR:
- Re-structured PayloadInspector base class
- Handling of IOV selection moved to the C++ client code
- Implemented new type of plots
- Adapted examples with BasicPayload
- Adapted existing plots for BEamSpot and SiStrip

### CODE REVIEW

**Comment by ggovi** [comment]
please test

**Comment by ggovi** [comment]
+1

**Comment by arunhep** [comment]
+1

File: CondFormats/Common/interface/BasicPayload.h [hunk 1]
```diff
@@ -9,14 +9,28 @@ namespace cond {
...
+	  std::cout <<ind<<":"<<m_vec[ind]<<"  ";
+	}
+	std::cout <<std::endl;
+      }
+    }
     
   public:
     float m_data0;
     float m_data1;
+    std::vector<float> m_vec;
```
Comment (davidlange6) [inline CondFormats/Common/interface/BasicPayload.h]: Hi @ggovi - is there a reason this isn't just a std::array<float,100>?

**Comment by davidlange6** [comment]
-1 
pending answer to comment(s)

File: CondFormats/Common/interface/BasicPayload.h (hunk 1 above)
Comment (ggovi) [inline CondFormats/Common/interface/BasicPayload.h]: @davidlange6 
Ok, point taken. This was only an example of simple payload. The fixed sized has been relaxed.

**Comment by ggovi** [comment]
please test

**Comment by ggovi** [comment]
+1

**Comment by arunhep** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Pull request merged**